### PR TITLE
chore: Move dependencies to a workspace `Cargo.toml` file

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

Addresses https://github.com/noir-lang/noir/issues/629 but for this repository

# Description

## Summary of changes

I've pulled up the shared dependencies into the workspace `Cargo.toml` file.

## Dependency additions / changes

N/A

## Test additions / changes

N/A

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,3 +1,23 @@
 [workspace]
 
 members = ["acir_field", "acir", "acvm", "stdlib"]
+
+[workspace.package]
+version = "0.4.1"
```
Comment (phated) [inline Cargo.toml:6]: This breaks some work I'm doing so we can automated releases for each package.

**Review by phated — DISMISSED** [review]
Can we just move deps and not the package information?

File: Cargo.toml (hunk 1 above)
Comment (TomAFrench) [reply Cargo.toml:6]: No worries, I assumed that we were keeping these in sync (as they currently are)

**Comment by TomAFrench** [comment]
> Can we just move deps and not the package information?

Do you just need the version to be crate-specific? Stuff like license, authors, etc. shouldn't affect automated releases.

**Comment by phated** [comment]
> Do you just need the version to be crate-specific? Stuff like license, authors, etc. shouldn't affect automated releases.

Correct, I just want the version in each package.

**Review by kevaundray — APPROVED** [review]

**Pull request merged**